scripts/imgtool/main.py

```diff
@@ -277,7 +279,21 @@ def sign(key, align, version, header_size, pad_header, slot_size, pad,
             # FIXME
             raise click.UsageError("Signing and encryption must use the same "
                                    "type of key")
-    img.create(key, enckey, dependencies)
+
+    # Get list of custom protected TLVs from the command-line
+    custom_tlv = {}
+    for tlv in cptlv:
+        tag = int(tlv[0], 16) if tlv[0].startswith('0x') else int(tlv[0])
+        if tag in custom_tlv:
+            raise ValueError('Custom TLV %s already exists.' % hex(tag))
+        elif tag in image.TLV_VALUES.values():
+            raise ValueError('Custom TLV %s conflicts with predefined TLV.' % hex(tag))
+        else:
+            length = tlv[1]
+            value = tlv[2]
+            custom_tlv[tag] = struct.pack(length, int(value))
```

Comment:
I am not entirely sure what you are trying to achieve here, but this is probably incorrect, what you should do is pass a tuple like this `custom_tlvs[tag] = (length, value)` and leave packing to be done in the creation routine which can deal with endianness.

Reply:
The idea here was to specify the length in python struct format: 'B' for byte, 'I' for uint.
Example: `--custom-tlv '0x81' 'I' '0xAABBCCDD'`

However, now I see another two ways:
 - specify length as integer `--custom-tlv '0x81' '4' '0xAABBCCDD'`
 - do not specify the length at all and get get it from the value length
In both cases try to convert value to numeric type if there is 0x prefix, otherwise convert to char array (`s` format from the https://docs.python.org/3.7/library/struct.html)